cmd/archive.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/driusan/dgit/git"
+)
+
+func argsParseCompressionLevel(opts *git.ArchiveOptions, args []string) []string {
+	for i, arg := range args {
+		if len(arg) < 2 || arg[0] != '-' {
+			continue
+		}
+		startIndex := 1
+		if arg[1] == '-' {
+			startIndex++
+			if len(arg) == 2 {
+				continue
+			}
+		}
+
+		// slice the dash
+		value := arg[startIndex:]
+
+		// Try to convert to integer
+		if v, err := strconv.Atoi(value); err != nil {
+			// The value is NaN
+			continue
+		} else {
+			// The compression value must be between 0..9
+			if v >= 0 && v <= 9 {
+				opts.CompressionLevel = v
+			}
+			// remove this value from the arg list
+			return append(args[:i], args[i+1:]...)
+		}
+	}
+	return args
+}
+
+func Archive(c *git.Client, args []string) error {
+	flags := flag.NewFlagSet("archive", flag.ExitOnError)
+	flags.SetOutput(flag.CommandLine.Output())
+	flags.Usage = func() {
+		flag.Usage()
+		fmt.Fprintf(flag.CommandLine.Output(), "\n\nOptions:\n")
+		flags.PrintDefaults()
+	}
+
+	opts := git.ArchiveOptions{}
+
+	//flags.BoolVar(&opts.Verbose, "verbose", false, "Report archived files on stderr")
+	//flags.BoolVar(&opts.Verbose, "v", false, "Alias for --verbose")
+	flags.Var(newNotimplBoolValue(), "verbose", "Not implemented")
+	flags.Var(newNotimplBoolValue(), "v", "Not implemented")
+
+	flags.StringVar(&opts.BasePrefix, "prefix", "", "Prepend prefix to each pathname in the archive")
+	flags.StringVar(&opts.OutputFile, "output", "", "Write the archive to this file")
+	flags.StringVar(&opts.OutputFile, "o", "", "Alias for --output")
+
+	//flags.BoolVar(&opts.WorktreeAttributes, "worktree-attributes", false, "Not implemented")
+	flags.Var(newNotimplBoolValue(), "worktree-attributes", "Not implemented")
+
+	flags.BoolVar(&opts.List, "list", false, "List supported archive formats")
+	flags.BoolVar(&opts.List, "l", false, "Alias for --list")
+
+	flags.Var(newNotimplStringValue(), "remote", "Not implemented")
+	flags.Var(newNotimplStringValue(), "exec", "Not implemented")
+
+	flags.StringVar(&opts.ArchiveFormat, "format", "", "Archive format")
+
+	flags.IntVar(&opts.CompressionLevel, "0", 0, "Store only")
+	flags.IntVar(&opts.CompressionLevel, "9", 9, "Highest compression level")
```

Comment:
You're missing 1-8 here (the man page only lists 0 and 9 explicitly, but the description for 9 says "You can specify any number from 1 to 9 to adjust compression speed and ratio.")

Reply:
They are parsed in the [argsParseCompressionLevel](https://github.com/driusan/dgit/blob/4ccf41a07eedc0902e948cf6f607489e6849cdb6/cmd/archive.go#L13) function outside of the go flags to avoid doing command line usage a lot verbose, for example official git client only shows options 0 and 9.
though i can change it to use the FlagSet if you want